tools/src/wyvern/tools/typedAST/core/expressions/Application.java

```diff
@@ -141,22 +160,31 @@ public FileLocation getLocation() {
 	public Expression generateIL(GenContext ctx, ValueType expectedType) {
 		CallableExprGenerator exprGen = function.getCallableExpr(ctx);
 		List<FormalArg> formals = exprGen.getExpectedArgTypes(ctx);
-		
+
+        int offset = 0;
 		// generate arguments		
 		List<Expression> args = new LinkedList<Expression>();
+        if(!this.generics.isEmpty()) {
+            for(String generic : this.generics) {
+                 args.add(new wyvern.target.corewyvernIL.expression.New(new TypeDeclaration(DefDeclaration.GENERIC_MEMBER, new NominalType("", generic), this.location)));
```

Comment:
I added a tool error for this situation, because the other options, (like "assume that the member should be named `T`", or "ignore the extra generic", or "default back to using just __generic__ without the extra letter") all seemed like brittle and illogical solutions.